src/Lecture4.hs

```diff
@@ -180,7 +219,24 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
+-- <!> defines a strict version of <> for the Maybe type.
+(<!>) :: Semigroup a => Maybe a -> Maybe a -> Maybe a
+Nothing   <!> Nothing   = Nothing
+(Just !x) <!> Nothing   = Just x
+Nothing   <!> (Just !y) = Just y
+(Just !x) <!> (Just !y) = Just (x <> y)
```

Comment:
That's absolutely correct implementation 👍🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
a <!> b = case a <> b of
    Nothing -> Nothing
    Just !x -> x
```

It's also a bit lazier but in a good way 👼🏻